[SYCL] Disable persistent device code caching by default. (#3891)
intel/llvm: sycl/doc/EnvironmentVariables.md

| `SYCL_ENABLE_PCI` | Integer | When set to 1, enables obtaining the GPU PCI address when using the Level Zero backend. The default is 0. |
| `SYCL_HOST_UNIFIED_MEMORY` | Integer | Enforce host unified memory support or lack of it for the execution graph builder. If set to 0, it is enforced as not supported by all devices. If set to 1, it is enforced as supported by all devices. |
| `SYCL_CACHE_DIR` | Path | Path to persistent cache root directory. Default values are `%AppData%\libsycl_cache` for Windows and `$XDG_CACHE_HOME/libsycl_cache` on Linux, if `XDG_CACHE_HOME` is not set then `$HOME/.cache/libsycl_cache`. |
| `SYCL_CACHE_TRACE` | Any(\*) | If the variable is set, messages are sent to std::cerr when caching events or non-blocking failures happen (e.g. unable to access cache item file). |

> | `SYCL_CACHE_DISABLE_PERSISTENT (deprecated)` | Any(\*) | Has no effect. |

Reviewer: Does it make sense to rework this var in similar to SYCL_HOST_UNIFIED_MEMORY, which is basically bool?
Call it SYCL_CACHE_PERSISTENT. Make it zero = OFF and non-zero = ON. Default can change independently based on what we decide now and later?
Looks like better control over behavior and more stable as a variable.

Author: addressed in 2964f91

> | `SYCL_CACHE_PERSISTENT` | Any(\*) | Controls persistent device compiled code cache - setting to '1' turns it on, to '0' - off. When on, SYCL runtime will try to cache and reuse JIT-compiled binaries. Default is off. |

Reviewer: ```suggestion
| `SYCL_CACHE_PERSISTENT` | Integer | Controls persistent device compiled code cache. Turns it on if set to '1' and turns it off if set to '0'. When cache is enabled SYCL runtime will try to cache and reuse JIT-compiled binaries. Default is off. |
```
